pkg/build/nix_generic.go

```diff
@@ -0,0 +1,79 @@
+package build
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"reflect"
+
+	"github.com/testground/testground/pkg/api"
+	"github.com/testground/testground/pkg/rpc"
+)
+
+var (
+	_ api.Builder = &NixBuilder{}
+)
+
+// NixBuilder (id: "nix:generic") is a builder that build the test plan using
+// Nix. The resulting artifact can be a binary or a container.
+type NixBuilder struct{}
+
+type NixBuilderConfig struct {
+	// AttributeToBuild is the attribute we're building with `nix build .#<attr>`.
+	// If empty, this will build the default package. e.g. `nix build .`
+	AttributeToBuild string `toml:"attr"`
+}
+
+// Build builds a testplan written in Go and outputs an executable.
```

Comment:
Am I right in saying it builds whatever `flake.nix` specifies so it doesn't necessarily have to be Go?